### PR TITLE
Get values after `_get_observation_pairs`

### PR DESCRIPTION
**Pull request by not522**
## Motivation
`_get_observation_pairs` computes the params and values information simultaneously, which makes the process difficult to follow. This PR changes it so that the params are obtained after `_get_observation_pairs`.

The name and documentation for `_get_observation_pairs` will not correspond to the implementation, but this will be corrected in a later PR.

## Description of the changes
- Get values after `_get_observation_pairs`

### CODE REVIEW

File: tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py [hunk 1]
```diff
@@ -311,7 +311,8 @@ def test_multi_objective_sample_independent_ignored_states() -> None:
 )
 @pytest.mark.parametrize("objective_value", [-5.0, 5.0, 0.0, -float("inf"), float("inf")])
 @pytest.mark.parametrize("multivariate", [True, False])
-@pytest.mark.parametrize("constant_liar", [True, False])
+# TODO(not522): Test constant_liar=True
```
Comment (HideakiImamura) [inline tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py:314]: Why does not this test `constant_liar=True`?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have a question.

File: tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py (hunk 1 above)
Comment (not522) [reply tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py]: I fixed the test as I realized we should ignore the `constant_liar` option as documented.

Comment (HideakiImamura) [reply tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py]: I see. Thanks for the explanation.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update. LGTM.

**Review by contramundum53 — APPROVED** [review]
LGTM!
A nit: Maybe the name `_get_observation_pairs` is obsolete, because we don't get the parameter values anymore?

**Comment by not522** [comment]
Yes, I will work on a later PR.

> The name and documentation for _get_observation_pairs will not correspond to the implementation, but this will be corrected in a later PR.

**Pull request merged**